docs/tutorials/introduction.rst

```diff
@@ -0,0 +1,355 @@
+``meson-python``.
+
+
+.. code-block:: toml
+   :caption: pyproject.toml
+
+   [build-system]
+   build-backend = 'mesonpy'
+   requires = ['meson-python']
+
+
+After we specify which backend to use, we'll want to define the package
+metadata. This is done in the ``project`` section, and the format is pretty
+self-explanatory by looking at our example.
+
+
+.. code-block:: toml
+   :caption: pyproject.toml
+
+   ...
+
+   [project]
+   name = 'our-first-project'
+   version = '0.0.1'
+   description = 'Our first Python project, using meson-python!'
+   readme = 'README.md'
+   requires-python = '>=3.8'
+   license = {file = 'LICENSE.txt'}
+   authors = [
+     {name = 'Bowsette Koopa', email = 'bowsette@example.com'},
+   ]
+
+
+.. admonition:: Adding dependencies
+   :class: seealso
+
+   If you need to add dependencies to the project metadata, you can check our
+   :ref:`how-to-guides-add-dependencies` guide.
+
+
+.. admonition:: Declaring project metadata
+   :class: seealso
+
+   Our example doesn't make use of all the fields available, so we recommend you
+   check out the `PyPA documentation on project metadata`_.
+
+
+.. admonition:: Writing TOML
+   :class: seealso
+
+   If you are not familiar with the TOML configuration language or need some
+   help, be sure to check out `toml.io <https://toml.io>`_.
+
+
+Testing the project
+-------------------
+
+Now we should have a valid Python project, let's test it.
+
+We will install it with pip_
+
+
+.. code-block:: console
+
+   $ pip install .
+   $ pip list
+   ...
+   our-first-project   0.0.1
+   ...
+
+
+After this, we should be able to import and try out our module.
+
+
+.. code-block:: console
+
+   $ python
+   >>> import our_first_module
+   >>> our_first_module.foo()
+   'bar'
+
+
+
+Creating your first release
+===========================
+
+Now that we have a valid Python project, we can release it.
+
+To release the project we first need to generate the distribution artifacts,
+these are files in a standardized format that Python package installers
+understand. There are two kind of artifacts, `source distributions`_, which are
+commonly referred to as *sdists*, and `binary distributions`_, which use a
+custom format named *wheel*, so they're generally referred to as *wheels*.
+
+
+What are the roles of *sdists* and *wheels*
+-------------------------------------------
+
+
+As you might have figured out by the name, *sdists* contain the source code of
+the project, and *wheels* contain a compiled [#pure-wheels]_ version of the
+project, ready to be copied to the file-system.
+
+If your project has compiled code, its *wheels* will be platform-dependent, and
+if it has native Python modules, the *wheels* will also be Python version
+specific, as the native Python modules depend on the Python ABI.
+FIXME(this seems *a bit* convoluted, it isn't awful, but I feel like it could
+be simplified or explained in a more natural way)
```

Comment:
This still needs addressing. I'd start with the common case:
```
If your project uses Python extension modules, your *wheels* will be specific to
both the platform and the Python version (footnote: unless you use CPython's stable ABI).
```

I would not even bother with a case where there is native code, but no extension modules - this is quite unlikely to happen (it'd require using `ctypes` or `cffi` to do anything with the native code).